### PR TITLE
Add index troubleshooting steps

### PR DESCRIPTION
**Pull request by andyw8**
Closes https://github.com/Shopify/ruby-lsp/issues/2296

### CODE REVIEW

File: jekyll/troubleshooting.markdown [hunk 1]
```diff
@@ -230,6 +230,18 @@ ruby-lsp
 Is there any extra information given from booting the server manually? Or does it only fail when booting through the
 extension?
 
+## Indexing
+
+When Ruby LSP starts, it attempts to index your code as well as your dependencies as described in [Configuring code indexing](index.html#configuring-code-indexing).
```
Comment (vinistock) [inline jekyll/troubleshooting.markdown:235]:
```suggestion
When Ruby LSP starts, it attempts to index your code as well as your dependencies as described in [Configuring code indexing](index#configuring-code-indexing).
```

File: jekyll/troubleshooting.markdown [hunk 2]
```diff
@@ -230,6 +230,18 @@ ruby-lsp
...
+
+In rare cases, Ruby LSP will encounter an error which prevents indexing from completing, which will result in incomplete information in the editor.
+
+Firstly, ensure that you are using the latest release of the `ruby-lsp` gem, as the problem may have been already fixed.
+
+To diagnose the particular file(s) causing a problem, run `ruby-lsp-check`. Please log an issue so that we can address it. If the code is not open source then please provide a minimal reproduction.
+
+In the meantime, you can [configure Ruby LSP to ignore a particular gem or file for indexing](index.html#configuring-code-indexing).
```
Comment (vinistock) [inline jekyll/troubleshooting.markdown:243]:
```suggestion
In the meantime, you can [configure Ruby LSP to ignore a particular gem or file for indexing](index#configuring-code-indexing).
```

**Review by vinistock — COMMENTED** [review]
We should add a link to the new documentation from the error dialog https://github.com/Shopify/ruby-lsp/pull/2367#pullrequestreview-2200331033

**Comment by andyw8** [comment]
Added link to error:

Screenshot 2024-09-30 at 10 51 10 AM

**Review by vinistock — APPROVED** [review]
Nice!

**Pull request merged**